src/debug.c

```diff
@@ -470,15 +472,16 @@ void debugCommand(client *c) {
 NULL
         };
         addReplyHelp(c, help);
-    } else if (!strcasecmp(c->argv[1]->ptr,"segfault")) {
+    } else if (!strcasecmp(c->argv[1]->ptr,"segfault") && c->argc == 2) {
```

Comment:
i rather not add strict syntax checks in debug command.
basically just because it doesn't really add any real value, and i rather not cause unnecessary merge and cherry pick conflicts.

please note that there is even a value of not checking for excessive arguments (even in real commands, not just in DEBUG).
such a case for instance is when you add an additional argument, if you know that the previous version ignored excess arguments, you can just send the new argument always, without the overhead of version check or a fallback to the old format.